dns-lookup-action.kamelet.yaml

```diff
@@ -0,0 +1,50 @@
+    camel.apache.org/provider: "Apache Software Foundation"
+  labels:
+    camel.apache.org/kamelet.type: "action"
+spec:
+  definition:
+    title: "DNS Lookup Action"
+    description: |-
+      Lookup for a domain
+
+      The Kamelet expects the following headers to be set:
+
+      - `domain-name` / `ce-domain-name`: as the domain for which we are looking up
+
+      If the domain-name header won't be set the body will be used as domain name.
+  types:
+    out:
+      mediaType: application/json
+  dependencies:
+  - "camel:jackson"
+  flow:
+    from:
+      uri: "kamelet:source"
+      steps:
+      - choice:
+          when:
+          - simple: "${header[domain-name]}"
+            steps:
+            - set-header:
+                name: dns.domain
+                simple: "${header[domain-name]}"
+          - simple: "${header[ce-domain-name]}"
+            steps:
+            - set-header:
+                name: dns.domain
+                simple: "${header[ce-domain-name]}"
+          otherwise:
+            steps:
+            - set-header:
+                name: dns.domain
+                simple: "${body}"
+      - marshal:
```

Comment:
Seems that marshal and to are in reversed order, isn't it?

Reply:
Fixed. Wrong commit pushed.